tests/prodtest_tests/__init__.py

```
@@ -0,0 +1,75 @@
+    The device formats binary payloads as an even-length run of uppercase hex
+    digits (see `cli_ok_hexdata`). This checks the args are valid, non-empty hex
+    and — when *num_bytes* is given — of the expected length, then returns the
+    decoded bytes for any further, non-fragile structural checks. It deliberately
+    does not compare the value itself, which is device-specific.
+    """
+    args = response.args
+    assert args, "expected hex data, got empty response"
+    try:
+        data = bytes.fromhex(args)
+    except ValueError:
+        raise AssertionError(f"response is not valid hex data: {args!r}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)